frontend/src/component/common/util.ts

```diff
@@ -98,11 +98,12 @@ export function updateWeightEdit(
     if (variants.length === 0) {
         return [];
     }
-    const { remainingPercentage, variableVariantCount } = variants.reduce(
+    let { remainingPercentage, variableVariantCount } = variants.reduce(
         ({ remainingPercentage, variableVariantCount }, variant) => {
             if (variant.weight && variant.weightType === weightTypes.FIX) {
                 remainingPercentage -= Number(variant.weight);
-            } else {
+            }
+            if (variant.weightType === weightTypes.VARIABLE) {
```

Comment:
there is was a bugfix here - it was not counting correctly when `weight == 0`

Reply:
do we need let?